### PR TITLE
we implemented a comment serializer

### PR DESCRIPTION
**Pull request by ghost**
This is our comment serializer

### CODE REVIEW

**Comment by therese28** [comment]
https://www.loom.com/share/9d672e66c3774f608c8456c1594dd966

**Comment by RipNasty** [comment]
Worked with Therese on this.  I'm curious if there's a different way to implement it other than a Promise but... it's the only way we could figure out how to make it work.

**Comment by AngeC22** [comment]
Looks good

**Comment by Chepenik** [comment]
Looks good to me. Think there might be a fair discussion to be had if we need ID in the comment serializer but I don't think it hurts if people can see what number this comment is. Great job on this pr:)

File: server/src/db/migrations/20230504135943_addImageUrlColumn.cjs [hunk 1]
```diff
@@ -7,7 +7,7 @@
  */
 exports.up = async (knex) => {
     return knex.schema.alterTable("websites", (table) => {
-        table.string("imageUrl").notNullable().defaultTo(null)
+        table.string("imageUrl").notNullable().defaultTo("")
```
Comment (nwalberts) [inline server/src/db/migrations/20230504135943_addImageUrlColumn.cjs:10]: i believe i saw this in the other PR. Doesnt make sense to default this to an empty string if its required.
remove the `defaultTo` altogether

File: server/src/serializers/WebsiteSerializer.js [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
         let serializedWebsite = {}
         for (const attribute of allowedAttributes) {
             serializedWebsite[attribute] = website[attribute]
         }
-        serializedWebsite.comments = await website.$relatedQuery("comments")
+        const relatedComments = await website.$relatedQuery("comments")
+        const serializedComments = await Promise.all(
+            relatedComments.map(async (comment) =>
+                CommentSerializer.showCommentDetails(comment))
+        )
```
Comment (nwalberts) [inline server/src/serializers/WebsiteSerializer.js:14]: pretty sure we talked about this and explicitly discussed how we _don't_ need Promise.all! remove the promise.all AND the `async` in your map

**Review by nwalberts — CHANGES_REQUESTED** [review]

**Pull request merged**